public/app/features/dashboard/components/DashNav/DashNav.tsx

```diff
@@ -57,6 +57,7 @@ type Props = OwnProps & ConnectedProps<typeof connector>;
 class DashNav extends PureComponent<Props> {
   constructor(props: Props) {
     super(props);
+    this.props.dashboard.meta.canShare = false;
```

Comment:
I also just wanted to leave a general comment to talk about where I'm coming from. In the design doc for our feature for containerizing Grafana, there was a lot of interest and push to try to upstream changes where possible. I told Alejandro and others that we would try to do that where we can. Pushing for upstreaming things helps me make good on my word to Alejandro, and it also reduces our squad's long-term maintenance burden for the fork.

That being said, I'm not the person with the difficult job of making these particular changes. It's easy for me to sit here and ask for things to be done in a generic way that we can upstream, but I recognize that it's hard to actually do that. Please feel free to push back and/or dismiss my comments, especially if a generic implementation adds a lot of effort.

Reply:
I definitely understand the rationale behind wanting to do things in a generic way. I think I will have to evaluate on a case-by-case basis. If it is straightforward enough to try to upstream a particular change, there's no reason we shouldn't. It might just result in a series on smaller PR's, but that is probably preferred anyways. I'll make sure to provide rationale for anything I choose not to upstream to make sure the team is onboard as well.